tutorials/performance/using_multiple_threads.rst

```diff
@@ -47,6 +47,91 @@ To create a thread, use the following code:
     func _exit_tree():
         thread.wait_to_finish()
 
+ .. code-tab:: cpp C++ .H File
+
+    #ifndef MULTITHREADING_DEMO_H
+    #define MULTITHREADING_DEMO_H
+    
+    #include <godot_cpp/classes/node.hpp>
+    #include <godot_cpp/classes/thread.hpp>
+    
+    namespace godot {
+    	class MultithreadingDemo : public Node {
+    		GDCLASS(MultithreadingDemo, Node);
+    
+    	private:
+    		Ref<Thread> worker;
+    
+    	protected:
+    		static void _bind_methods();
+    		void _notification(int p_what);
+    
+    	public:
+    		MultithreadingDemo();
+    		~MultithreadingDemo();
+    
+    		void demo_threaded_function();
+    	};
+    }
+
+    #endif // MULTITHREADING_DEMO_H
+
+ .. code-tab:: cpp C++ .CPP File
+
+    #include "multithreading_demo.h"
+
+    #include <godot_cpp/classes/engine.hpp>
+    #include <godot_cpp/classes/time.hpp>
+    #include <godot_cpp/core/class_db.hpp>
+    #include <godot_cpp/variant/utility_functions.hpp>
+    
+    using namespace godot;
+    
+    void MultithreadingDemo::_bind_methods() {
+    	// THREADING
+    	ClassDB::bind_method(D_METHOD("threaded_function"), &MultithreadingDemo::demo_threaded_function);
+    }
+    
+    void MultithreadingDemo::_notification(int p_what) {
+    	// Prevents this from running in the editor, only during game mode
+    	if (Engine::get_singleton()->is_editor_hint()) {
+    		return;
+    	}
+    
+    	switch (p_what) {
+    		case NOTIFICATION_READY: {
+    			worker.instantiate();
+    			worker->start(callable_mp(this, &MultithreadingDemo::demo_threaded_function), Thread::PRIORITY_NORMAL);
+    		} break;
+    		case NOTIFICATION_EXIT_TREE: {
+    			if (worker.is_valid() && worker->is_alive()) {
+    				worker->wait_to_finish();
+    			}
+    
+    			worker->unreference();
+    		};
+    	}
+    }
+    
+    MultithreadingDemo::MultithreadingDemo() {
+    	// Initialize any variables here.
+    }
+    
+    MultithreadingDemo::~MultithreadingDemo() {
+    	// Add your cleanup here.
+    }
+    
+    void MultithreadingDemo::demo_threaded_function() {
+    	UtilityFunctions::print("demo_threaded_function started!");
+    	int i = 0;
+    	uint64_t start = Time::get_singleton()->get_ticks_msec();
+    	while (Time::get_singleton()->get_ticks_msec() - start < 5000) {
+    		i++;
```

Comment:
This is not a recommended function IMO, it counts up, with no delays, for half a second, it will overflow and cause all sorts of weirdness, I'd suggest something else, something with a delay preferably, like:
```suggestion
    		OS::get_singleton()->delay_msec(10);
    		i++;
```

Otherwise it looks good style wise

Reply:
I'm speaking for "coder" from the forums, but I believe the idea is if you test this with two threads, you get different results to see a result of multithreading. I have that test working in another file.

Would it work to keep this, but declare it as an unsigned int and break the while loop if it's about to overflow?

Reply:
Not really a good idea, we shouldn't suggest people running code like that IMO, having a busy loop like this isn't good, so just adding a delay is useful, and a delay wouldn't change that, but it should really be some useful code here

Reply:
Oops I misread what you changed -- totally makes sense!